src/discussions/comments/CommentsView.jsx

```diff
@@ -3,23 +3,35 @@ import React, { useEffect } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
 import { useParams } from 'react-router';
 
+import { ensureConfig, getConfig } from '@edx/frontend-platform';
 import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
 import { Button, Spinner } from '@edx/paragon';
 
 import { selectThread } from '../posts/data/selectors';
+import { markThreadAsRead } from '../posts/data/thunks';
 import Post from '../posts/post/Post';
 import { selectThreadComments } from './data/selectors';
 import { fetchThreadComments } from './data/thunks';
 import Reply from './reply/Reply';
 import messages from './messages';
 
+ensureConfig(['POST_MARK_AS_READ_DELAY'], 'Comment thread view');
+
 function CommentsView({ intl }) {
   const { postId } = useParams();
   const dispatch = useDispatch();
   const thread = useSelector(selectThread(postId));
   const comments = useSelector(selectThreadComments(postId));
   useEffect(() => {
     dispatch(fetchThreadComments(postId));
+    const markReadTimer = setTimeout(() => {
+      if (thread && !thread.read) {
+        dispatch(markThreadAsRead(postId));
+      }
+    }, getConfig().POST_MARK_AS_READ_DELAY);
+    return () => {
+      clearTimeout(markReadTimer);
+    };
```

Comment:
if we only have to dispatch one call under set timeout which is conditional, then can we shift the if statement above like this?

`if (thread && !thread.read) {
        const markReadTimer = setTimeout(() => {
            dispatch(markThreadAsRead(postId));
        }, getConfig().POST_MARK_AS_READ_DELAY);
        return () => {
          clearTimeout(markReadTimer);
        };
}`

Reply:
I think the code got a bit jumbled above, but if I understand the intention right you'd like to make the setTimeout and return conditional? 

The  issue there is that either react or the linter doesn't like this and recommends having a return for both paths, which mean we'd need a additional bogus return statement which makes the code less clean.

Reply:
yes, my intention here is that if we don't need to dispatch the call then put all the code under condition.  I am not sure about your statement here. but I am good with it if you feel so.